linchpin/__init__.py

```diff
@@ -122,6 +122,8 @@ def runcli(ctx, config, workspace, verbose, version, creds_path):
 
     if workspace is not None:
         ctx.workspace = os.path.realpath(os.path.expanduser(workspace))
+    else:
```

Comment:
@tvieira #320 does not remove the self.workspace setting from LinchPinContext. Can you maybe clarify what you mean wrt this? I'm reworking a few things here to address the bug above, so I'd like your input. Thanks!